### ISSUE TITLE
unsoundness in  RomHandle::new

### ISSUE THREAD

**Issue by lwz23** [RWTH-OS/eduOS-rs#46]
I notice this https://github.com/RWTH-OS/eduOS-rs/blob/d7bc227f169a65cd3f7cc12d818662ee69d8f061/src/fs/initrd.rs#L31
consider the following code:
```
pub fn new(addr: *const u8, len: usize) -> Self {
		RomHandle {
			pos: Spinlock::new(0),
			data: Arc::new(RwLock::new(unsafe { slice::from_raw_parts(addr, len) })),
		}
	}
```

**Bug Description**
The new function in the provided code is unsound because it uses the unsafe function slice::from_raw_parts without validating its arguments. The function assumes that the caller provides:

A valid pointer (addr) pointing to an allocated memory region.
A valid length (len) that does not exceed the bounds of the memory region.
A memory region that outlives the created RomHandle.
If any of these assumptions are violated, the behavior of slice::from_raw_parts is undefined, leading to potential memory safety issues.

**Consequences of the Bug**
Null Pointer Dereference: If addr is a null pointer, dereferencing it during slice creation will cause undefined behavior, such as a segmentation fault.
Out-of-Bounds Access: If len exceeds the actual size of the allocated memory region, accessing the slice can result in out-of-bounds memory reads or writes.
Dangling Pointer Access: If the memory region pointed to by addr is deallocated or does not outlive the RomHandle, accessing the slice later will cause use-after-free errors or other undefined behavior.

**PoC**
```
fn main() {
    let invalid_ptr = std::ptr::null();
    let len = 10;

    let _rom_handle = RomHandle::new(invalid_ptr, len); // UB
}
```
**Consequences of the POC**
Null Pointer: The null pointer invalid_ptr is passed to slice::from_raw_parts, which expects a valid memory address. This causes undefined behavior immediately when the slice is created.
Segmentation Fault: On many systems, attempting to dereference a null pointer results in a segmentation fault, crashing the program.
Undefined Behavior: Even if the program does not immediately crash, further access to the data field of RomHandle will result in unpredictable behavior, including memory corruption or program instability.

**Comment by lwz23** [comment]
ping?

**Comment by stlankes** [comment]
You are right! I am currently rewriting the kernel. Currently I am working on stage6. If I start to rewrite stage7, I will try to remove  this behavior.

**Comment by stlankes** [comment]
Commit 98e5ca3540263d90f00d4ee73651e1ee1e08b7b5 should solve this issue.
